src/main/java/com/bytecoders/pharmaid/PharmaidController.java

```diff
@@ -153,6 +190,59 @@ public ResponseEntity<?> addPrescription(@PathVariable("userId") String userId,
     }
   }
 
+  /**
+   * Endpoint to update a user's prescriptions.
+   *
+   * @param userId user whose prescriptions we're trying to retrieve
+   * @param prescriptionId  prescription we are trying to update
+   * @param request containg prescription update
+   * @return a ResponseEntity with updated prescription if the operation is successful,
+   *      or an error message if an error occurred
+   */
+  @PatchMapping(path = "/users/{userId}/prescriptions/{prescriptionId}")
+  public ResponseEntity<?> updateUsersPrescription(
+      @PathVariable(USER_ID) String userId,
+      @PathVariable(PRESCRIPTION_ID) String prescriptionId,
+      @RequestBody @Valid UpdatePrescriptionRequest request) {
+    try {
+      final Optional<User> userOptional = userService.getUser(userId);
+
+      if (userOptional.isEmpty()) {
+        return new ResponseEntity<>("Provided User does not exist", HttpStatus.NOT_FOUND);
+      }
+
+      final Optional<Prescription> prescriptionOptional =
+            prescriptionService.getPrescriptionById(prescriptionId);
+
+      if (prescriptionOptional.isEmpty() 
+          || !prescriptionOptional.get().getUser().getId().equals(userId)) {
+        return new ResponseEntity<>(
+          "Provided Prescription doesn't exist",
+          HttpStatus.NOT_FOUND);
+      }
+
+      Prescription prescription = prescriptionService.getPrescriptionById(prescriptionId).get();
+
+      if (request.getEndDate() != null) {
+        prescription.setEndDate(request.getEndDate());
+      }
```

Comment:
Right now, we're only able to update endDate, but we should be able to update:
- dosage
- numOfDoses
- startDate
- endDate